vision_agent/tools/meta_tools.py

```diff
@@ -659,8 +675,21 @@ def use_object_detection_fine_tuning(
     ]
 
     new_code = code
-    for pattern, replacer in patterns:
-        new_code = re.sub(pattern, replacer, new_code)
+
+    for index, (pattern_with_fine_tune_id, replacer_with_fine_tune_id) in enumerate(
+        patterns_with_fine_tune_id
+    ):
+        if re.search(pattern_with_fine_tune_id, new_code):
+            new_code = re.sub(
+                pattern_with_fine_tune_id, replacer_with_fine_tune_id, new_code
+            )
+        else:
+            (pattern_without_fine_tune_id, replacer_without_fine_tune_id) = (
+                patterns_without_fine_tune_id[index]
```

Comment:
a lit bit confused here:
If the first `re.search` on 682 failed, here we just use the index 0?
What if the `new_code` is `owl_v2_image`?